### DISCUSSION TITLE
Uploading a wordpress site created in Laragon with the correct web address

### DISCUSSION THREAD

**Discussion by Anthony1070** [leokhoa/laragon#466] [Laragon's Articles & How-tos]
Hi,

I need you help.

I have a website ( https://www.webfoodculture.com ) and, since it's quite old, I want to rebuild its pages from scratch using Elementor.

So, I have installed Wordpress locally using Laragon (btw, it's really friendly!). In WP, I have installed Elementor and created the new pages.

I want to upload them, replacing the old website with the new one, but before doing that I have to be certain that every single page has the same address of the one is going to be replaced.

In order to do this I have changed in Largon preferences the hostname template from  www.{name}.dev.com to www.{name}.com (where "name" is webfooculture. (img01)
![img_01](https://user-images.githubusercontent.com/130087639/230455075-a0de9353-0b3a-4405-b73c-3b1dd6199dbf.jpg)

The URL of the local pages changes accordingly (I have also activated SSL, so I get "https://" )   (img01)
![img_02](https://user-images.githubusercontent.com/130087639/230457727-c984de37-ad5e-4728-939d-5efcf7173877.jpg)

Once all the pages have the correct address, I will backup the local site (I use Updraft Plus plugin), and I will restore it online.

Do you think this is the right procedure correct?

I need to know this before ruining weeks of hard work.

Many thanks in advance for your help!

Anthony

**Comment by Andres-M-Rios** [comment]
Hello Antonio,

 

I just subscribed to the email chain from this development project. I am a long time Web Engineer, so you are almost there, just going to give you a few tips and tricks from a place of many years of experience doing stuff with pre-existing websites and new builds.

 

Step One:

Back up the current version of the site, all HTML and CSS and JS and whatever other files, plus any current database that is part of any built in CMS on the website. (Is it an old WordPress, or it’s much older than that??) (Is there any type of CMS on the old site??)

 

Step One (B):

Also might want to make sure you have everything from the Export step (from Up Draft Plug In) in WordPress saved to one ZIP file to transfer to the new site.

 

Stop Two:

Plan Plan Plan. You want to minimize any down time, and you want it to be at a time when your user counts are lowest, for example Midnight where ever the majority of your users are.

 

Step Three:

Build a blank WordPress shell on your current hosting setup. Most hosts will have something ease to use on the host, like scriptaculous, or another “Web Site Builder” web App once you login to your account. That’s the best way to start the process, so no settings from your home development system will cause headaches.

 

Stop Four:

Okay, now on the online WordPress, make sure the same Plugins are installed as your development system. (Or at least all the ones you made use of). Then import everything from that ZIP file to the Server on the web. (Your Hosted WordPress Solutions).

 

As long as you follow these steps, it should be fine, and, if you hit a snag, you can simply go back to the backup you did at step one, restore the old site, and troubleshoot what went wrong, figure out exactly what the problem was and come up with a plan of attack.

 

Then when you can solve whatever the snag was, start over at step one with attempt #2 with no major outage of the site.

 

Hopefully you can get it done in one attempt, as I think you have a good understanding of the tools, and any other specific clarification you still need please just email me directly at ***@***.***

 

Good luck!

++++

Andres M. Partridge-Rios

 

From: Antonio ***@***.***> 
Sent: Thursday, April 6, 2023 2:02 PM
To: leokhoa/laragon ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [leokhoa/laragon] Uploading a wordpress site created in Laragon with the correct web address (Discussion #466)

 

Hi,

I need you help.

I have a website ( https://www.webfoodculture.com ) and, since it's quite old, I want to rebuild its pages from scratch using Elementor.

So, I have installed Wordpress locally using Laragon (btw, it's really friendly!). In WP, I have installed Elementor and created the new pages.

I want to upload them, replacing the old website with the new one, but before doing that I have to be certain that every single page has the same address of the one is going to be replaced.

In order to do this I have changed in Largon preferences the hostname template from www.{name}.dev.com <http://www.%7Bname%7D.dev.com>  to www.{name}.com <http://www.%7Bname%7D.com>  (where "name" is webfooculture. (img01)
 <https://user-images.githubusercontent.com/130087639/230455075-a0de9353-0b3a-4405-b73c-3b1dd6199dbf.jpg> 

The URL of the local pages changes accordingly (I have also activated SSL, so I get "https://" ) (img01)
 <https://user-images.githubusercontent.com/130087639/230457727-c984de37-ad5e-4728-939d-5efcf7173877.jpg> 

Once all the pages have the correct address, I will backup the local site (I use Updraft Plus plugin), and I will restore it online.

Do you think this is the right procedure correct?

I need to know this before ruining weeks of hard work.

Many thanks in advance for your help!

Anthony

**Comment by Anthony1070** [comment]
Hello Andres!

Many many thanks for your fast reply (mine, sadly, was not so quick.

It seems you're telling me that I'm following the right path building the local "offline" new site (I am still quite worried about the pages address part),

I want just to ask you about "Step One A": yes, my actual site (webfoodculture.com) is made Wordpress, I count to use Updraft to backup it, but since the plugin may have problems (it happened in the past), can you suggest me the bast way to backup it manually?

About the transfer, please keep in mind that I'm going to move to a new hosting provider (right now I'm on NameHero, I'll probably choose Hostinger): my idea was keep the original site for same days, setup the new site on the new host and, once everything is ok, ask to the techs of the new provider to change the DNS so that the switch will be completely "transparent" to the visitors. What do you think?

Again, thanks a lot for your help, it's much appreciated!

Ant

**Comment by dontletthesamedogbiteyoutwice** [comment]
Just go here [https://www.akeeba.com/products/akeeba-backup-wordpress.html](url)
That's the free version. Use it to backup your existing site. Also install it on your Laragon site.
I've been using it for years

**Reply by Anthony1070** [reply]
Thanks a lot for your suggestion dontlet. Much appreciated.
A.
